core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -354,9 +362,10 @@ object Runtime extends RuntimePlatformSpecific {
       def shutdown()(implicit unsafe: Unsafe): Unit
     }
 
-    override def unsafe: UnsafeAPI with UnsafeAPI2 with UnsafeAPI3 =
-      new UnsafeAPIV2 {}
-
+    override def unsafe: UnsafeAPI with UnsafeAPI2 with UnsafeAPI3 = {
+      if (unsafe0 eq null) unsafe0 = new UnsafeAPIV2 {}
```

Comment:
Don't we need to cache this? We could unsafely set the type of `unsafe0` to `Object` (which is what `UnsafeAPI with UnsafeAPI2 with UnsafeAPI3` is compiled to anyways)